mmdeploy/codebase/mmyolo/deploy/object_detection.py

```diff
@@ -0,0 +1,261 @@
+            tuple: (data, img), meta information for the input image and input.
+        """
+
+        from mmcv.transforms import Compose
+        if not isinstance(imgs, (list, tuple)):
+            imgs = [imgs]
+        dynamic_flag = is_dynamic_shape(self.deploy_cfg)
+        cfg = process_model_config(self.model_cfg, imgs, input_shape)
+        # Drop pad_to_square when static shape. Because static shape should
+        # ensure the shape before input image.
+
+        pipeline = cfg.test_pipeline
+        if not dynamic_flag:
+            transform = pipeline[1]
+            if 'transforms' in transform:
+                transform_list = transform['transforms']
+                for i, step in enumerate(transform_list):
+                    if step['type'] == 'Pad' and 'pad_to_square' in step \
+                       and step['pad_to_square']:
+                        transform_list.pop(i)
+                        break
+        test_pipeline = Compose(pipeline)
+        data = []
+        for img in imgs:
+            # prepare data
+            if isinstance(img, np.ndarray):
+                # TODO: remove img_id.
+                data_ = dict(img=img, img_id=0)
+            else:
+                # TODO: remove img_id.
+                data_ = dict(img_path=img, img_id=0)
+            # build the data pipeline
+            data_ = test_pipeline(data_)
+            data.append(data_)
+
+        data = pseudo_collate(data)
+        if data_preprocessor is not None:
+            data = data_preprocessor(data, False)
+            return data, data['inputs']
+        else:
+            return data, BaseTask.get_tensor_from_input(data)
+
+    @staticmethod
+    def get_partition_cfg(partition_type: str) -> Dict:
+        """Get a certain partition config for mmdet.
+
+        Args:
+            partition_type (str): A string specifying partition type.
+
+        Returns:
+            dict: A dictionary of partition config.
+        """
+        from .model_partition_cfg import MMYOLO_PARTITION_CFG
+        assert (partition_type in MMYOLO_PARTITION_CFG), \
+            f'Unknown partition_type {partition_type}'
+        return MMYOLO_PARTITION_CFG[partition_type]
+
+    def get_preprocess(self) -> Dict:
+        """Get the preprocess information for SDK.
+
+        Return:
+            dict: Composed of the preprocess information.
+        """
+        input_shape = get_input_shape(self.deploy_cfg)
+        model_cfg = process_model_config(self.model_cfg, [''], input_shape)
+        preprocess = model_cfg.test_pipeline
+        return preprocess
+
+    def get_postprocess(self) -> Dict:
+        """Get the postprocess information for SDK.
+
+        Return:
+            dict: Composed of the postprocess information.
+        """
+        postprocess = self.model_cfg.model.test_cfg
+        if 'rpn' in postprocess:
+            postprocess['min_bbox_size'] = postprocess['rpn']['min_bbox_size']
+        if 'rcnn' in postprocess:
```

Comment:
is there any instance segmentation model in mmyolo? if not, could remove mask related configs